beats/libbeat/scripts/generate_template.py to support empty "nested" or "object" type definitions

Currently it is not possible to configure a type "nested" or "object" for a field but don't have any child fields defined. Currently this leads to an error.
This would be useful when the the child fields are not known in advance and/or are that many that you don't want to explicitly map them but would like to fall back to dynamic templates for the unmapped fields.

Here an example what I want to achieve:
[fields.txt](https://github.com/elastic/beats/files/699775/fields.txt)

The complete source code is available at: https://github.com/christiangalsterer/httpbeat
The interesting part is the jsonBody field. It contains an arbitrary ison structure which depends on the HTTP endpoint. Therefore no fixed mapping can be defined. My idea was here to use object/nested.

Currently this is generated:
`
"jsonBody": {
              "ignore_above": 1024,
              "type": "keyword"
            }
`

But I would like this to be generated:
`
"jsonBody": {
                "properties": {}
            }
`
When I use 
`
        - name: jsonBody
          type: nested
          description: >
            HTTP response body returned by the URL endpoint in JSON format.
`

I get the following error:
`
Traceback (most recent call last):
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 327, in <module>
    fields_to_es_template(args, fields, output, args.beatname + "-*", version_data['version'])
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 98, in fields_to_es_template
    defaults, "")
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 161, in fill_section_properties
    prop, dynamic = fill_field_properties(args, field, defaults, path)
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 263, in fill_field_properties
    prop, dynamic = fill_section_properties(args, field, defaults, path)
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 161, in fill_section_properties
    prop, dynamic = fill_field_properties(args, field, defaults, path)
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 276, in fill_field_properties
    prop, dynamic = fill_section_properties(args, field, defaults, path)
  File "./vendor/github.com/elastic/beats/libbeat/scripts/generate_template.py", line 160, in fill_section_properties
    for field in section["fields"]:
KeyError: 'fields'
make: *** [update] Error 1
`

--- a/libbeat/scripts/generate_template.py
+++ b/libbeat/scripts/generate_template.py
@@ -162,10 +162,11 @@ def fill_section_properties(args, section, defaults, path):
     properties = {}
     dynamic_templates = []
 
-    for field in section["fields"]:
-        prop, dynamic = fill_field_properties(args, field, defaults, path)
-        properties.update(prop)
-        dynamic_templates.extend(dynamic)
+    if "fields" in section:
+        for field in section["fields"]:
+            prop, dynamic = fill_field_properties(args, field, defaults, path)
+            properties.update(prop)
+            dynamic_templates.extend(dynamic)
 
     return properties, dynamic_templates
 
@@ -292,12 +293,12 @@ def fill_field_properties(args, field, defaults, path):
             path = field["name"]
         prop, dynamic = fill_section_properties(args, field, defaults, path)
 
+        properties[field.get("name")] = {
+            "type": "nested",
+            "properties": {}
+        }
         # Only add properties if they have a content
         if len(prop) is not 0:
-            properties[field.get("name")] = {
-                "type": "nested",
-                "properties": {}
-            }
             properties[field.get("name")]["properties"] = prop
 
         dynamic_templates.extend(dynamic)

